source/PlayerInfo.cpp

```diff
@@ -3672,6 +3672,15 @@ void PlayerInfo::RegisterDerivedConditions()
 	pluginProvider.SetHasFunction(pluginFun);
 	pluginProvider.SetGetFunction(pluginFun);
 
+	auto &&destroyedPersonProvider = conditions.GetProviderPrefixed("person destroyed: ");
+	auto destroyedPersonFun = [](const string &name) -> bool
+	{
+		const Person *person = GameData::Persons().Get(name.substr(strlen("person destroyed: ")));
+		return person ? person->IsDestroyed() : false;
```

Comment:
`Set::Get` won't return null pointers, it'll create a new object in the collection if one with this name doesn't exist.

```suggestion
		const Person *person = GameData::Persons().Find(name.substr(strlen("person destroyed: ")));
		return person ? person->IsDestroyed() : false;
```